vpr/src/draw/search_bar.cpp

```diff
@@ -127,7 +127,7 @@ void search_and_highlight(GtkWidget* /*widget*/, ezgl::application* app) {
 
 bool highlight_rr_nodes(int hit_node) {
     t_draw_state* draw_state = get_draw_state_vars();
-
+    auto& device_ctx = g_vpr_ctx.device();
```

Comment:
I don't think you change anything in this context, so better to mark it as const
const auto& device_ctx = g_vpr_ctx.device();

Reasons:
1. Makes it more clear to someone reading the code that you won't modify this (large) global construct.
2. Can help the compiler optimize, by guaranteeing you will never modify this object through this reference/pointer. This can help it determine if it is safe to keep some of the data accessed through this pointer in registers, rather than always reloading them by accessing main memory via the pointer (i.e. it simplifies pointer alias analysis).